src/libraries/Microsoft.PowerFx.Core/Syntax/Nodes/DecLitNode.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax.SourceInformation;
+
+namespace Microsoft.PowerFx.Syntax
+{
+    /// <summary>
+    /// Decimal literal parse node. Example:
+    /// 
+    /// <code>12.34</code>
+    /// </summary>
+    public sealed class DecLitNode : TexlNode
+    {
+        // If Value is non-null, then the token represents its value.
+        // Otherwise, the value is in DecValue.
+        internal readonly decimal DecValue;
```

Comment:
this seems confusing. Why shouldn't it always be in DecValue?

Reply:
I took this from NumLitNode which has the same comments and logic.  The only real difference between the two is that NumLitNode uses NaN and DecLitNode uses 0 as the default in the constructor, since decimal dos not have a NaN.